### PR TITLE
INTYGFV-15980: Conversion of line separators to br tag in HTMLFactory.

### PR DESCRIPTION
**Pull request by ChrilleSigerhed**

### CODE REVIEW

**Comment by ChrilleSigerhed** [comment]
image

File: app/src/main/java/se/inera/intyg/minaintyg/util/html/HTMLFactory.java [hunk 1]
```diff
@@ -19,8 +21,23 @@ public static String tag(String tagName, String className, String value) {
...
+    return value;
+  }
+
+  private static String convertLineSeparators(String value) {
+    return value.replace(LINE_SEPARATOR, BR_TAG);
+  }
 
-    return startTag(tagName, className) + value + endTag(tagName);
+  private static boolean hasLineSeparators(String value) {
+    return value.contains("\n");
```
Comment (mww-sopra-steria) [inline app/src/main/java/se/inera/intyg/minaintyg/util/html/HTMLFactory.java:40]: Use constant.

File: app/src/main/java/se/inera/intyg/minaintyg/util/html/HTMLFactory.java [hunk 2]
```diff
@@ -10,6 +10,8 @@ public class HTMLFactory {
   private static final String START_ATTRIBUTE_TAG = "=\"";
   private static final String END_ATTRIBUTE_TAG = "\"";
   private static final String SPACE = " ";
+  public static final String LINE_SEPARATOR = "\n";
```
Comment (mww-sopra-steria) [inline app/src/main/java/se/inera/intyg/minaintyg/util/html/HTMLFactory.java:13]: Make constants private as they are just part of this class.

File: app/src/main/java/se/inera/intyg/minaintyg/util/html/HTMLFactory.java [hunk 3]
```diff
@@ -19,8 +21,23 @@ public static String tag(String tagName, String className, String value) {
     if (value == null || tagName == null || tagName.isEmpty()) {
       return "";
     }
+    final var text = convertLineSeparatorsIfPresent(value);
+    return startTag(tagName, className) + text + endTag(tagName);
+  }
+
+  private static String convertLineSeparatorsIfPresent(String value) {
```
Comment (mww-sopra-steria) [inline app/src/main/java/se/inera/intyg/minaintyg/util/html/HTMLFactory.java:28]: It is fine to keep as is... but it is a bit redundant to first check for LINE_SEPARATOR and then replace. The <string>.replace(...) will do just that by itself.

**Review by mww-sopra-steria — APPROVED** [review]
Nice! Minor comments on code that you could look at before merging!

**Pull request merged**